src/Highway.Data/Interfaces/IDataContextBase.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Linq;
+
+namespace Highway.Data
+{
+    /// <summary>
+    ///     Contract that provides <see cref="IQueryable" /> instances.
+    /// </summary>
+    public interface IDataContextBase
+    {
+        /// <summary>
+        ///     This gives a mock-able wrapper around normal Set method that allows for testability
+        /// </summary>
+        /// <typeparam name="T">The Entity being queried</typeparam>
+        /// <returns>
+        ///     <see cref="IQueryable{T}" />
+        /// </returns>
+        IQueryable<T> AsQueryable<T>()
+            where T : class;
+    }
+}
```

Comment:
This interface exists as the single common type between IDataContext and IReadonlyDataContext.  This seems like the cleanest way to allow IReadonlyDataContext to function with the existing queries, scalars, and commands.  I also the following approaches, but I prefer this one:

- Use a common base interface for IDataContext / IReadonlyDataContext **(THIS)**
- Make IReadonlyDataContext inherit from IDataContext  and throw an exception in non-implemented members.  Seems awfully tacky.
- Add overloads to the Execute, Find, Etc. methods in query/scalar classes.
- replace IDataContext with IReadonlyDataContext in query/scalar classes and have DataContext inherit from ReadonlyDataContext.  But now we'd be back to that situation we all disliked where IDataContext inherits from IReadonlyDataContext, which just doesn't read well.  "I'm derived from the readonly thing, but I can write?  Hmm..."